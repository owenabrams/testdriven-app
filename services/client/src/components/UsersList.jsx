import React from 'react';

const UsersList = (props) => {
  return (
    <div>
      {
        props.users.map((user) => {
          return (
            <div
              key={user.id}
              className="box"
            >
              <div className="level">
                <div className="level-left">
                  <h4 className="title is-4 mb-1">
                    {user.username}
                  </h4>
                </div>
                <div className="level-right">
                  {/* Sync Status Indicator */}
                  {user.synced ? (
                    <span className="tag is-success is-small">
                      ✅ Synced
                    </span>
                  ) : (
                    <span className="tag is-warning is-small">
                      ⏳ Pending
                    </span>
                  )}
                </div>
              </div>

              <p className="subtitle is-6 has-text-grey">
                {user.email}
              </p>

              {/* Additional metadata for debugging */}
              {user.createdAt && (
                <p className="is-size-7 has-text-grey-light">
                  Created: {new Date(user.createdAt).toLocaleString()}
                </p>
              )}
            </div>
          )
        })
      }

      {/* Empty state */}
      {props.users.length === 0 && (
        <div className="has-text-centered has-text-grey">
          <p className="is-size-5">No users yet</p>
          <p className="is-size-6">Add your first user above!</p>
        </div>
      )}
    </div>
  )
};

export default UsersList;
