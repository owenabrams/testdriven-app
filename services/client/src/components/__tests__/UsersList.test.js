import React from 'react';
import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';

import UsersList from '../UsersList';

const users = [
  {
    'active': true,
    'email': 'hermanmu@gmail.com',
    'id': 1,
    'username': 'michael'
  },
  {
    'active': true,
    'email': 'michael@mherman.org',
    'id': 2,
    'username': 'michaelherman'
  }
];

test('UsersList renders properly', () => {
  const { container } = render(<UsersList users={users}/>);
  const userBoxes = container.querySelectorAll('.box');
  const userTitles = container.querySelectorAll('h4');
  expect(userBoxes.length).toBe(2);
  expect(userTitles.length).toBe(2);
  expect(userTitles[0].textContent).toBe('michael');
  expect(userTitles[1].textContent).toBe('michaelherman');
});

test('UsersList renders a snapshot properly', () => {
  const tree = renderer.create(<UsersList users={users}/>).toJSON();
  expect(tree).toMatchSnapshot();
});
