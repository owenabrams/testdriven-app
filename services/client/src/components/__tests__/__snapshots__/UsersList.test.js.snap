// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UsersList renders a snapshot properly 1`] = `
<div>
  <div
    className="box"
  >
    <div
      className="level"
    >
      <div
        className="level-left"
      >
        <h4
          className="title is-4 mb-1"
        >
          michael
        </h4>
      </div>
      <div
        className="level-right"
      >
        <span
          className="tag is-warning is-small"
        >
          ⏳ Pending
        </span>
      </div>
    </div>
    <p
      className="subtitle is-6 has-text-grey"
    >
      hermanmu@gmail.com
    </p>
  </div>
  <div
    className="box"
  >
    <div
      className="level"
    >
      <div
        className="level-left"
      >
        <h4
          className="title is-4 mb-1"
        >
          michaelherman
        </h4>
      </div>
      <div
        className="level-right"
      >
        <span
          className="tag is-warning is-small"
        >
          ⏳ Pending
        </span>
      </div>
    </div>
    <p
      className="subtitle is-6 has-text-grey"
    >
      michael@mherman.org
    </p>
  </div>
</div>
`;
